examples/contrib/detection/faster_rcnn.py

```diff
@@ -0,0 +1,286 @@
+                    x["name"],
+                ),
+                annotation["object"],
+            )
+        )
+        if self.transforms is not None:
+            result = self.transforms(image=img, bboxes=bbox_classes)
+        image, bbox_classes = result["image"] / 255.0, result["bboxes"]
+        bboxes = np.stack([a[:4] for a in bbox_classes])
+        labels = [self.name2class[a[4]] for a in bbox_classes]
+
+        target = {}
+        target["boxes"] = torch.tensor(bboxes)
+        target["labels"] = torch.tensor(labels)
+        target["image_id"] = annotation["filename"]
+        target["area"] = torch.tensor((bboxes[:, 3] - bboxes[:, 1]) * (bboxes[:, 2] - bboxes[:, 0]))
+        target["iscrowd"] = torch.tensor([False] * len(bboxes))
+
+        return image, target
+
+
+def collate_fn(batch):
+    return tuple(zip(*batch))
+
+
+def run(
+    experiment_name,
+    dataset_root: str,
+    log_dir: str,
+    epochs: int,
+    batch_size: int,
+    lr: int,
+    download: bool,
+    device: str,
+    image_size: int,
+) -> None:
+    bbox_params = A.BboxParams(format="pascal_voc")
+    train_transform = A.Compose(
+        [A.Resize(image_size, image_size), A.HorizontalFlip(p=0.5), ToTensorV2()],
+        bbox_params=bbox_params,
+    )
+    val_transform = A.Compose([A.Resize(image_size, image_size), ToTensorV2()], bbox_params=bbox_params)
+
+    train_dataset = Dataset(root=dataset_root, download=download, image_set="train", transforms=train_transform)
+    val_dataset = Dataset(root=dataset_root, download=download, image_set="val", transforms=val_transform)
+    vis_dataset = Subset(val_dataset, random.sample(range(len(val_dataset)), k=16))
+
+    train_dataloader = DataLoader(
+        train_dataset, batch_size=batch_size, shuffle=True, collate_fn=collate_fn, num_workers=4
+    )
+    val_dataloader = DataLoader(val_dataset, batch_size=batch_size, shuffle=False, collate_fn=collate_fn, num_workers=4)
+    vis_dataloader = DataLoader(vis_dataset, batch_size=batch_size, shuffle=False, collate_fn=collate_fn, num_workers=4)
+
+    model = fasterrcnn_resnet50_fpn(pretrained=True)
+    in_features = model.roi_heads.box_predictor.cls_score.in_features
+    model.roi_heads.box_predictor = FastRCNNPredictor(in_features, 21)
+    model.to(device)
+
+    if device == "cuda":
+        scaler = GradScaler()
+    optimizer = SGD(lr=lr, params=model.parameters())
+    scheduler = OneCycleLR(optimizer, max_lr=lr, total_steps=len(train_dataloader) * epochs)
+
+    def update_model(engine, batch):
+        model.train()
+        images, targets = batch
+        images = list(image.to(device) for image in images)
+        targets = [{k: v.to(device) for k, v in t.items() if isinstance(v, torch.Tensor)} for t in targets]
+
+        if device == "cuda":
+            with torch.cuda.amp.autocast(enabled=True):
+                loss_dict = model(images, targets)
+                loss = sum(loss for loss in loss_dict.values())
```

Comment:
I think pytorch also support cpu autocast: https://pytorch.org/docs/stable/amp.html?highlight=autocast#torch.autocast
so we can unify the code...